### PR TITLE
Add provider and workspace create commands

### PR DESCRIPTION
**Pull request by cdupuis**
This PR add two new commands:

* `workspace create` creates a new workspace for the user
* `provider create` creates a new SCM provider for convergence by an SDM

Question: Should those commands be `create workspace` and `create provider` instead of the currently proposed names.

### CODE REVIEW

File: lib/config.ts [hunk 1]
```diff
@@ -178,15 +179,17 @@ async function createApiKey(cfg: Configuration): Promise<string> {
             name: "apiKey",
             transformer: maskString,
             message: `Enter your ${chalk.cyan("api key")} from ${chalk.yellow("https://app.atomist.com/apikeys")}
-    or hit ${chalk.cyan("<ENTER>")} to select an authentication provider to login with Atomist`,
+    or hit ${chalk.cyan("<ENTER>")} to select an authentication provider to login with Atomist
+    to login with Atomist:`,
```
Comment (ddgenome) [inline lib/config.ts]: ???

File: lib/config.ts [hunk 2]
```diff
@@ -178,15 +179,17 @@ async function createApiKey(cfg: Configuration): Promise<string> {
...
         },
     ];
 
     let answers = await inquirer.prompt(questions);
     if (!answers.apiKey) {
         const providers = await axios.get(`${cfg.endpoints.auth}/providers`);
 
-        print.log(`Select one of the following authentication providers available to login with Atomist:`);
+        print.log(`Select one of the following authentication providers
+available to login with Atomist:`);
```
Comment (ddgenome) [inline lib/config.ts]: Is this newline intentional? If so, it's cleaner to just use `\n`.

File: lib/provider.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import { resolveUserConfig } from "./cliConfig";
+import {
+    configureWorkspaces,
+    validateApiKey,
+} from "./config";
+import * as print from "./print";
+import { createGitHubCom } from "./provider/github";
+
+interface ProviderTypes {
+    [key: string]: {
```
Comment (ddgenome) [inline lib/provider.ts]: Could have used `type ProviderTypes = Record<string, {…}>;`

File: lib/provider.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+/*const UnsupportedProvider = async (workspaceId: string,
+                                   apiKey: string,
+                                   cfg: Configuration) => {
+    print.error("SCM provider not supported yet!");
+    return {
+        code: 1,
+        configuration: cfg,
+    };
+};*/
```
Comment (ddgenome) [inline lib/provider.ts]: Do you want to retain this comment?

File: lib/provider.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        create: UnsupportedProvider,
+    },
+    gitlab_com: {
+        label: "GitLab.com",
+        create: UnsupportedProvider,
+    },
+    bitbucket: {
+        label: "BitBucket",
+        create: UnsupportedProvider,
+    },*/
```
Comment (ddgenome) [inline lib/provider.ts]: Retain comment?

File: lib/provider/github.ts [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+    };
+}
+
+async function obtainToken(): Promise<string> {
+    let token: string;
+
+    print.log(`In order to create a GitHub SCM resource provider,
+we need to obtain a GitHub token with ${chalk.cyan("admin:org_hook")}
+scope. The token will ${chalk.bold("not")} be stored with Atomist!`);
+    const questions: inquirer.Question[] = [
```
Comment (ddgenome) [inline lib/provider/github.ts]: This all looks very familiar.

File: lib/provider/github.ts [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+            return {
+                code: 1,
+                configuration: {},
+            };
+        }
+    }
+
+    print.log(`Specify which GitHub organizations and/or repositories
+you want to connect to Atomist. Please comma separated
+names or glob patterns.`);
```
Comment (ddgenome) [inline lib/provider/github.ts]: "Please provide a comma separated list of organization names and/or glob patterns".

File: lib/workspace.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        return 1;
+    }
+
+    let workspaceId;
+    try {
+        workspaceId = await createWorkspace(opts.workspaceName, apiKey, cfg);
+    } catch (e) {
+        print.error(`Failed to create workspace: ${e.message}`);
+        return 1;
+    }
```
Comment (ddgenome) [inline lib/workspace.ts]: The above code is very similar to the first part of the create function in provider.ts.

File: lib/workspace.ts [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    createTeam: {
+        id: string;
+        name: string;
+    };
+}
+
+/**
+ * Command-line options and arguments for workspace create
+ */
+export interface CreateOptions {
```
Comment (ddgenome) [inline lib/workspace.ts]: This is very similar to the CreateOptions in provider.ts. Should they be distinguished by adding the type, i.e., CreateWorkspaceOptions and CreateProviderOptions? Maybe this and the `create` function is why you did the `import *` from both of these?

File: lib/config.ts (hunk 1 above)
Comment (cdupuis) [reply lib/config.ts]: argh, rebase issue. fixing, obviously.

File: lib/provider.ts (hunk 4 above)
Comment (cdupuis) [reply lib/provider.ts]: yes

**Pull request merged**